STIPS/STIP-009.md

```diff
@@ -0,0 +1,188 @@
+## Timeline
+|  Action               |  End Date  |
+|---                    |---         |
+| Technical Spec        |   2/25     |
+| Implementation        |   3/2      |
+| Auditors              |   3/11     |
+| Launch                |   3/18     |
+## Checkpoint 1
+Before more in depth design of the contract flows lets make sure that all the work done to this point has been exhaustive. It should be clear what we're doing, why, and for who. All necessary information on external protocols should be gathered and potential solutions considered. At this point we should be in alignment with product on the non-technical requirements for this feature. It is up to the reviewer to determine whether we move onto the next step.
+
+**Reviewer**: LGTM @bweick
+
+## Proposed Architecture Changes
+![](../assets/stip-009/image1.png "")
+
+**ManagerFactory**: Factory smart contract which enables the creation of new Set Tokens with a BaseManagerV3 manager, the migration of existing Set Tokens to a BaseManagerV3 manager, and the initialization of modules and extensions.
+
+**BaseManagerV3**: Manager smart contract which supports three roles (`owner`, `methodologist`, `operator`) and an asset whitelist which can be set by the `owner` to restrict the `operator`'s interactions with extensions. 
+
+**BasicIssuanceExtension**: Global extension which provides manager smart contracts with an interface to the BasicIssuanceModule (`issue`, `redeem`).
+
+**StreamingFeeExtension**: Global extension which provides manager smart contracts with an interface to the StreamingFeeModule (`accrueFeesAndDistribute`, `updateStreamingFee`, `updateFeeRecipient`, `updateFeeSplit`).
+
+**TradeExtension**: Global extension which provides manager smart contracts with an interface to the TradeModule (`trade`).
+
+## Requirements
+
+**ManagerFactory**
+
+- Allow owners to create new set tokens with a manager smart contract
+- Allow owners to migrate existing set tokens to a manager smart contract
+- Allow owners to create manager smart contract and initialize and parameterize all modules and extensions in two transactions
+
+**BaseManagerV3**
+
+- Allow owners to permission specific functionality (extensions) to different operators
```

Comment:
This kinda reads like an operator can be permissioned to one specific extension. Let's make that a little more clear, something like `owner can add and remove global operator permissions`